JIRA STORM-766 (Include version info in the service page)

redsanket opened the pull request:
I have included the code to display the version info in the web-page. Let me know if you need any other requirement.

Parth-Brahmbhatt commented on storm-core/src/clj/backtype/storm/converter.clj:
```diff
@@ -5,14 +5,17 @@
 
 (defn thriftify-supervisor-info [supervisor-info]
-  (doto (SupervisorInfo.)
+  (log-message "supervisor-version " (:version supervisor-info))
```
unnecessary log.

Parth-Brahmbhatt commented:
All the generated files that really has no change other then the date should be removed from the PR. Looks good otherwise, will test once you remove the files.

redsanket commented:
Hi Parth,
I have reverted back the unmodified thrift java classes and also removed the uneccessary logs. Kindly let me know if you need anything else to be done. Thanks

revans2 commented on storm-core/src/clj/backtype/storm/converter.clj:
```diff
@@ -23,7 +25,9 @@
       (if (.get_scheduler_meta supervisor-info) (into {} (.get_scheduler_meta supervisor-info)))
-      (.get_uptime_secs supervisor-info))))
+      (.get_uptime_secs supervisor-info)
+      (.get_version supervisor-info);;log
```
Please clean up the comment, not sure what ;;log means, and move the closing ')' to this line

revans2 commented on storm-core/src/storm.thrift:
```diff
@@ -153,6 +153,7 @@ struct SupervisorSummary {
   3: required i32 num_workers;
   4: required i32 num_used_workers;
   5: required string supervisor_id;
+  6: required string version;
```
This needs to be optional, to maintain rolling upgrade compatibility.

revans2 commented:
Overall the code looks good.  The trick with the rolling upgrade is that we want to be sure that an old nimbus can work with a new supervisor and a new nimbus can work with an old supervisor too.  Similarly for the client.  An old client should be able to work with a new nimbus and a new client should, in most cases, work with an old nimbus. Perhaps the simples way to do this is to provide a default value for version which is set to something like "VERSION NOT PROVIDED".

redsanket commented:
I have modified the code as per the specifications. Please let me know if there are any specifications that have to be met. Thanks Sanket

revans2 commented:
Please upmerge, there is a merge conflict in supervisor.clj, but it does not look too complex.

revans2 commented:
I ran a simple manual test where I had an old supervisor and a new nimbus and everything worked well.  I am +1 for merging this in once the merge conflict is resolved.

revans2 commented:
+1 looks good to me.
